infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java

```diff
@@ -98,16 +97,27 @@ protected OpenShiftEnvironment doCreate(
                 + "application/x-yaml, text/yaml, text/x-yaml");
     }
 
-    final KubernetesList list =
-        clientFactory.create().lists().load(new ByteArrayInputStream(content.getBytes())).get();
+    final KubernetesList list;
+    try {
+      list =
+          clientFactory.create().lists().load(new ByteArrayInputStream(content.getBytes())).get();
+    } catch (KubernetesClientException e) {
+      String message = e.getCause() == null ? e.getMessage() : e.getCause().getLocalizedMessage();
+      if (message.contains("\n")) {
+        // Clean up message if it comes from JsonMappingException. Format is e.g.
+        // `No resource type found for:v1#Route1\n at [...]`
+        message = message.split("\\n")[0];
```

Comment:
The same as in `KubernetesEnvironmentFactory`

Reply:
Done.